pipenv/patched/piptools/repositories/pypi.py

```diff
@@ -245,7 +258,10 @@ def get_legacy_dependencies(self, ireq):
                     setup_requires = self.finder.get_extras_links(
                         dist.get_metadata_lines('requires.txt')
                     )
-            except TypeError:
+                ireq.version = dist.version
+                ireq.project_name = dist.project_name
+                ireq.req = dist.as_requirement()
```

Comment:
I don’t quite understand why these are needed :|

Reply:
You just have to believe me that they are, based on errors I encountered

Reply:
Sometimes the `InstallRequirement` doesn't properly get these values set on it during the resolution process because of ephemeral wheel building moving on too quickly or something along those lines, it's quite hard to pin down exactly where in the stack it is going wrong which is why I added an additional safeguard to try and get the actual name of the package in question and resolve the actual requirement and version -- these elements are critical for downstream resolution if we want to provide _real_ conflict resolution